contracts/src/token/erc20/extensions/permit.rs

```diff
@@ -57,8 +57,12 @@ pub enum Error {
 sol_storage! {
     /// State of a Permit Contract.
     pub struct Permit<T: IEip712 + StorageType>{
```

Comment:
I think that the name in this case should be like `Erc20Permit` since rn it is tightly coupled to `Erc20` contract.